ansicolor.js

```diff
@@ -198,131 +213,220 @@ const TEXT    = 0,
       BRACKET = 1,
       CODE    = 2
 
-function rawParse (s) {
-    
-    let state = TEXT, buffer = '', text = '', code = '', codes = []
-    let spans = []
-
-    for (let i = 0, n = s.length; i < n; i++) {
-
-        const c = s[i]
+class Span {
+  constructor(code, text) {
+    this.code = code;
+    this.text = text;
+
+    // Those are added in the actual parse, this is done for performance reasons to have the same hidden class
+    this.css = "";
+    this.color = "";
+    this.bgColor = "";
+    this.bold = undefined;
+    this.inverse = undefined;
+    this.italic = undefined;
+    this.underline = undefined;
+    this.bright = undefined;
+    this.dim = undefined;
+  }
+}
 
-        buffer += c
+// getString as function instead of string to allow garbage collection
+function* rawParse(getString) {
+  const stateObject = {
+    state: TEXT,
+    buffer: "",
+    text: "",
+    code: "",
+    codes: [],
+  };
 
-        switch (state) {
+  const ONE_MB = 1048576;
 
-            case TEXT:
-                if (c === '\u001b') { state = BRACKET; buffer = c; }
-                else                { text += c }
-                break
+  const chunks = splitStringToChunksOfSize(getString(), ONE_MB);
```

Comment:
Split the string into chunks of 1MB to allow fast garbage collection of 1MB at a time - instead of holding the reference to the string we split into chunks of 1MB and after processing is finished we can remove the reference so it can be GC